test/lib/rubocop/iconbutton_component_test.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "lib/cop_test_case"
+
+class IconButtonMigrationTest < CopTestCase
+  def cop_class
+    ::RuboCop::Cop::Migrations::IconButtonComponent
+  end
+
+  def test_no_warnings_on_beta_icon_button
+    investigate(cop, <<~RUBY)
+      Primer::Beta::IconButton.new()
+    RUBY
+
+    assert_empty cop.offenses
+  end
+
+  def test_warn_on_icon_button_no_args
+    investigate(cop, <<~RUBY)
+      Primer::IconButton.new()
+    RUBY
+
+    assert_equal 1, cop.offenses.count
+  end
+
+  def test_warn_on_icon_button_simple_arg
+    investigate(cop, <<~RUBY)
+      Primer::IconButton.new(0)
+    RUBY
+
+    assert_equal 1, cop.offenses.count
+    assert cop.offenses.first.corrector.present?
+
+    assert_equal "Primer::Beta::IconButton.new(0)", cop.offenses.first.corrector.rewrite.strip
```

Comment:
I believe `Primer::Beta::IconButton.new(0)` will raise an error when executed because the constructor does not accept any positional arguments. The name of this test suggests the cop should warn in this case, not autocorrect - is that right?

Reply:
This invocation is more meant to just test that arguments are correctly transferred when correcting. The arguments are not altered, and this isn't a linter to check for correctness in the arguments, so I think this case is okay.

Reply:
Ah ok I see. I think I would feel better if the example used kwargs since that's much closer to how these instances will look in dotcom.

Reply:
Yes, that looks good to me!